### ISSUE TITLE
p215 this.foo vs this.get('foo')

### ISSUE THREAD

**Issue by edsnowden** [balinterdi/rarwe-issues#380]
I suspect this was covered in a previous chapter but I've not been able to find it...

The code on p215 says

`let searchTerm = this.searchTerm.toLowerCase();`

but then

`return this.model.get('songs').filter(...);`

I'm wondering why we don't need to do `this.get('searchTerm')` but we do need to do `this.model.get('songs')` instead of just `this.model.songs`. On p246, for example, you use `this.model.name` rather than `this.model.get('name')`.

Is it because 'searchTerm' is the dependency named by the first argument to computed(), and thus we can rely on this.searchTerm being up-to-date by the time our compute function is called?

Or is it because model is a distinct EmberObject (i.e. not the controller object) and thus we always need to call get() regardless of context? Or, is it because it's a model object and we need to call get() in case getting songs triggers an API call to fetch the data?

Trying to come up with some hard-and-fast rules, as I can see this is going to catch me out all the time when I come to writing my own code...

**Comment by balinterdi** [comment]
The fact that we have to use `get` for fetching Ember Data relationships is explained on page 147:

> By defining songs as a hasMany property, we tell Ember Data to put the Song objects belonging to a particular band in an array that can be accessed as band.get('songs'). The use of get instead of the standard JavaScript property accessor (band.songs) is one of the few places we still need to do this in Ember. The reason is that the Ember Data relationship returns a "promise proxy object" instead of a regular record array. To read more about this, check out the "EmberObject accessors" section in the Appendix.

This is the case since Ember 3.1 as explained in [the release blog post](https://emberjs.com/blog/2018/04/13/ember-3-1-released.html#toc_es5-getters-for-computed-properties-2-of-4).

tl;dr Using get is only needed in chained property lookups (e.g `user.get('address.shipping.zip')`, ED relationships and if you define `unknownProperty` on your class).
